jstests/aggregation/isodates.js

```diff
@@ -0,0 +1,243 @@
+/*
```

Comment:
While these tests are wonderful, and have good coverage, is there any way we could add them to [this test file](https://github.com/mongodb/mongo/blob/r3.3.2/jstests/aggregation/bugs/server6189.js)? If we need more coverage, I'd rather test all the other date operators as well. If not, I'd suggest adding some as unit tests in [this file](https://github.com/mongodb/mongo/blob/r3.3.2/src/mongo/db/pipeline/expression_test.cpp)